### ISSUE TITLE
timeout retry and resolve_retries in resolver section

### ISSUE THREAD

**Issue by ogrand** [cloudfoundry-incubator/haproxy-boshrelease#172]
In my haproxy configuration, I use dns resolver on legacy servers and I need to set 2 properties `timeout retry` and `resolve_retries` to manage these dns servers.
It should be very usefull to have these properties in resolver section (like `hold valid`).

**Comment by domdom82** [comment]
hi @ogrand can you add a little more detail on what you would like to see implemented exactly? do you want to add two more properties [here](https://github.com/cloudfoundry-incubator/haproxy-boshrelease/blob/master/jobs/haproxy/templates/haproxy.config.erb#L201)?

**Comment by ogrand** [comment]
hi @domdom82, my need is to define 2 haproxy resolvers properties `timeout retry` and `resolve_retries` as following

**jobs/haproxy/templates/haproxy.config.erb**
```
<% if_p("ha_proxy.resolvers") do |resolvers| -%>
resolvers default
    hold valid <%= p("ha_proxy.dns_hold") %>
    timeout retry <%= p("ha_proxy.timeout_retry") %>
    resolve_retries <%= p("ha_proxy.resolve_retries") %>
  <%- resolvers.each do |resolver| -%>
    nameserver <%= resolver.keys[0] %> <%= resolver.values[0] %>:53
  <%- end -%>
<% end -%>
```

**spec/haproxy_templates_spec.rb**
```
        'dns_hold' => '10s',
        'timeout_retry' => '1s',
        'resolve_retries' => 3,
```
**jobs/haproxy/spec**
```
  ha_proxy.dns_hold:
    description: "DNS Hold time"
    default: 10s
  ha_proxy.timeout_retry:
    description: "Timeouts related to name resolution"
    default: 1s
  ha_proxy.resolve_retries:
    description: "Number of dns queries to send to resolve a server name before giving up"
    default: 3
```
I can try to complete my initial PR https://github.com/cloudfoundry-incubator/haproxy-boshrelease/pull/171 if you want, but I don't want to make conflicts for **jobs/haproxy/templates/haproxy.config.erb** which is modified in previous PR (not merge).

**Comment by domdom82** [comment]
hi @ogrand that makes sense to me in a new PR. I have merged #171  as it covered all the fixes needed. Please go ahead and open a new PR. I will review and merge.
